Fix: Make actions in userhub clickable

davecreaser opened the pull request:
## Description

- Make items in the userhub clickable, to open the relevant action in the action sidebar
  - Stake items
  - Transaction items (if they have a relevant action hash)
- Update the UI of the userhub

Details here: https://github.com/JoinColony/colonyCDapp/issues/2261

## Testing

Check that the UI matches the designs: https://www.figma.com/design/l1dOM5qiQYwF0ElvKDqqjg/Design-System---Colony-v3?node-id=1782-50890&t=Zn39wYmkuOXiZ0z7-0

Check that stake items in the userhub are clickable, and open the action sidebar with the correct action.

Check that transaction items open the action sidebar with the correct action ONLY IF they have a hash which has a relevant action.

## Diffs

**Changes** 🏗

- Add buttons to the userhub items to create links to the relevant actions (transactions can now be expanded using the chevron button)
- Change html and classes of the userhub items to match the new designs

Resolves #https://github.com/JoinColony/colonyCDapp/issues/2261

melyndav reviewed (changes requested):
Hey @davecreaser nice start on this one! Just a couple of minor pick ups.

1. The HR between items isn't the right colour token.

<img width="532" alt="image" src="https://github.com/JoinColony/colonyCDapp/assets/82127904/39c45795-84c8-4bc8-9ed1-c9f7ce766695">
 
 2. Gap between title and date should be 8px not 6px wide as per figma below. 
 
<img width="526" alt="image" src="https://github.com/JoinColony/colonyCDapp/assets/82127904/bcb6a922-4160-4bfe-a04d-9ca7456ebf5b">

3. When the sidepanel is open and the user selects a transaction etc from the user hub, can we make the user hub close? It currently stays open once the transaction is loaded.

<img width="1079" alt="image" src="https://github.com/JoinColony/colonyCDapp/assets/82127904/7b8c2f8a-621f-47e5-91e0-f658e4f795db">

davecreaser commented:
Thanks @melyndav , I've addressed the issues :)

Also just an important note: not all transactions have an associated action, so we can't open the sidebar with an action for those transactions. I've made those transactions not clickable. Hope that makes sense!

rdig commented on src/types/transactions.ts:
```diff
@@ -70,3 +70,35 @@ export enum TRANSACTION_METHODS {
   Approve = 'approve',
 }
+
+export enum BatchKeys {
```
The batch keys are actually method names, so they should be merged with this: https://github.com/JoinColony/colonyCDapp/blob/185cae2e8073c2241e5425a2a50bd9886e6d68de/src/types/transactions.ts#L67C19-L72

rdig reviewed (approved):
Yup, all good here. Left a comment that needs addressing, but otherwise, nicely done! :100:

iamsamgibbs reviewed (changes requested):
Nice work on this ticket, all working nicely and some nice tidying up of the code.

Only one minor issue I've noticed, on the grouped transactions the pills and chevrons should be right aligned (I think they are already, but the rows aren't filling up the entire space, needs min-width: 100% or something).

<img width="674" alt="Screenshot 2024-05-16 at 09 27 13" src="https://github.com/JoinColony/colonyCDapp/assets/34915414/eac2f0fc-988b-44a5-8028-915b1a7d25c1">

melyndav reviewed (changes requested):
Hey @davecreaser thanks for the fixes.

In the empty state for transactions, could you kindly align the no results to match the Figma design? It's gone a bit awol. 

Same for stakes, could you also align the empty state in the centre of the content block?

<img width="745" alt="image" src="https://github.com/JoinColony/colonyCDapp/assets/82127904/b7d10347-5859-43e2-9571-250e27a30dee">

https://www.figma.com/design/l1dOM5qiQYwF0ElvKDqqjg/Design-System---Colony-v3?node-id=1782-51660&t=YgaJ7uYVBRvSyvEi-4

Also, confirming @iamsamgibbs comment regarding the alignment of the content also needs to be resolved on mobile.

<img width="555" alt="image" src="https://github.com/JoinColony/colonyCDapp/assets/82127904/67f78cd1-9e06-4a04-93a7-9e341df484db">

Also, within the stakes tab, can you align the pill to the center of the row (consistent with transactions) and also match the font colour of the stakes/ rewards/ penalty text to Figma as well (text-gray-500).

<img width="1462" alt="image" src="https://github.com/JoinColony/colonyCDapp/assets/82127904/27b85fee-ab36-4600-864d-cbfed6b3ab48">

melyndav reviewed (changes requested):
Hey @davecreaser thanks for your continued work on this!

A couple of refinements are needed in the stakes tab to handle some UI issues.

1. Mobile is displaying the staking data with weird wraps:
<img width="416" alt="image" src="https://github.com/JoinColony/colonyCDapp/assets/82127904/8f23c435-d9e2-4201-90f2-a393f429b743">

2. Desktop has some font weight issues and also padding issues with the staking data as well.
<img width="490" alt="image" src="https://github.com/JoinColony/colonyCDapp/assets/82127904/407abfb8-90f7-44b0-a5cd-82b02814edea">

Font weights should be normal as per figma:
<img width="1136" alt="image" src="https://github.com/JoinColony/colonyCDapp/assets/82127904/dfae2570-27fd-465c-92a9-e564d3560a59">
 
 There should be 12px padding between each data block as per figma:

<img width="1128" alt="image" src="https://github.com/JoinColony/colonyCDapp/assets/82127904/feec27ab-efba-4af7-82d3-fc34763ba782">

Also, I think the metadata is showing but in stakes we only want to show stake/reward/penalty data which is contributing to the wrap issue
<img width="673" alt="image" src="https://github.com/JoinColony/colonyCDapp/assets/82127904/1c544f7a-8323-426c-bf72-b8fab3fb522a">

3. There should only be a 8px padding within the transaction accordion to reduce the height of these sections.
<img width="424" alt="image" src="https://github.com/JoinColony/colonyCDapp/assets/82127904/03567059-ae8e-4acc-9174-507c97756bbd">

davecreaser commented:
Good eagle eyes @melyndav ! Thanks for the time reviewing. I think I've caught everything you've raised now, but let me know if you spot anything else :)

melyndav reviewed (approved):
Super appreciate all the extra work to fix these minor UI bits @davecreaser. Nice work as always 💯
